tests/cogs/test_token_remover.py

```diff
@@ -0,0 +1,139 @@
+import asyncio
+import logging
+import unittest
+from unittest.mock import MagicMock
+
+from discord import Colour
+
+from bot.cogs.token_remover import (
+    DELETION_MESSAGE_TEMPLATE,
+    TokenRemover,
+    setup as setup_cog,
+)
+from bot.constants import Channels, Colours, Event, Icons
+from tests.helpers import AsyncMock
+
+
+class TokenRemoverTests(unittest.TestCase):
+    """Tests the `TokenRemover` cog."""
+
+    def setUp(self):
+        """Adds the cog, a bot, and a message to the instance for usage in tests."""
+        self.bot = MagicMock()
+        self.bot.get_cog.return_value = MagicMock()
+        self.bot.get_cog.return_value.send_log_message = AsyncMock()
+        self.cog = TokenRemover(bot=self.bot)
+
+        self.msg = MagicMock()
```

Comment:
Probably a good idea to create a mock helper for messages and use mocks with specs for some of the attributes here (like the author and channel).

Reply:
That seems reasonable to me.

I think we should factor out the mocking of those Discord "Model" that we use a lot and typically require the same or a very similar set-up. Since it does mean we'd basically have to maintain and test a small `discord.py` mocking library, I'd say we should restrict it those those base models though. I think that with a set of Member, Guild, Role, Message, TextChannel, Context, Bot, and Role we've covered a lot of things that we typically usr.

Reply:
@MarkKoz @jchristgit I'm going to add a couple more specific mocks and tests, but I'll resolve this during the merge.

Reply:
Do this means this should be left alone for now?